rtengine/ipwavelet.cc

```diff
@@ -3191,7 +3192,9 @@ void ImProcFunctions::ContAllL (float *koeLi[12], float *maxkoeLi, bool lipschit
             float scale = 1.f;
             float scale2 = 1.f;
 
-            float LL100, LL100res, LL100init, kH[maxlvl];
```

Comment:
Which Coverity defect is intended to be fixed here? Did you target this one?
````
CID 157589 (#1 of 1): Uninitialized scalar variable (UNINIT)271. uninit_use: Using uninitialized value kH[level].
````
This at least is not solved by allocating the memory dynamically which will only slow things down and lead to new bugs if it isn't freed properly.

Also the Coverity report is a bit tricky here IMHO: It tells us that if `maxlvl == 0`, the variable `kH` will never be set because the for loop initializing it will never be entered. But then again it will be an array of length zero and there will be nothing to initialize inside. In summary, I think the correct fix would be to ensure that the parameters `level` and `maxlvl` make sense, e.g. fulfill relations like `0 <= level < maxlvl` immediately upon entering the function.